### ISSUE TITLE
WU upload fails `HTTP_NOT_ACCEPTABLE: {"error":{"message":"Quit","code":406}}`

### ISSUE THREAD

**Issue by Hou5e** [FoldingAtHome/fah-client-bastet#72]
First running of client, the WU upload fails, and probably a couple other scenarios like: restarting the client, starting again after 'finish and pause' (I think).

The upload fails, and there is no retry.  You are not credited with the points for the WU, and the WU is not accepted (for the WU in the log file below): https://apps.foldingathome.org/wu#project=18449&run=0&clone=306&gen=291

Example log file: [log.txt](https://github.com/FoldingAtHome/fah-client-bastet/files/10308222/log.txt)

WUs after this seem to work well and don't have this problem (at least on Linux for GPU WUs that I've tested about 100 good and only the first one bad).

Dumping the first (CPU) WU will also likely have this problem, and completing the next (GPU) WU will also likely have this problem.

This seems to have to do with the client's initial conditions each time it starts up, and probably makes the return package of files with a mistake when it sends them to the collection server.
It happened in v8.1.5, and v8.1.6 for me on Windows or Linux.

```
2022-12-27:11:48:30:I1::WU1:Folding@home Core Shutdown: FINISHED_UNIT
2022-12-27:11:48:31:I1::WU1:Core returned FINISHED_UNIT (100)
2022-12-27:11:48:31:I1::WU1:Uploading WU results
2022-12-27:11:48:32:I1:OUT10:> POST https://vav19.fah.temple.edu/api/results HTTP/1.1
2022-12-27:11:48:32:I3:Connecting to vav19.fah.temple.edu:443
2022-12-27:11:52:35:I1:OUT10:< vav19.fah.temple.edu:443 HTTP/1.1 406 HTTP_NOT_ACCEPTABLE
2022-12-27:11:52:35:E ::WU1:HTTP_NOT_ACCEPTABLE: {"error":{"message":"Quit","code":406}}
```

**Comment by Hou5e** [comment]
The fastest way (at least on Windows) to reproduce this error is to start with a blank Data directory (or just contains config.xml), then install or run FAH Client, and dump the first CPU that auto-starts (I let it run for a bit). Sending the WU dumping results back fails with this error. I have the debug output for that (let me know if you need more, I can run a WU and get an error 406 that way too in about 6-8 hours on this Windows PC).

**Comment by Hou5e** [comment]
The debug results have your passkey in them, file password 'jeffslogfile':
[log-20221228-064113-DebugHTTP_Error406.zip](https://github.com/FoldingAtHome/fah-client-bastet/files/10314813/log-20221228-064113-DebugHTTP_Error406.zip)
(I'll probably delete this comment later today)

**Comment by jcoffland** [comment]
Thanks!

**Comment by Hou5e** [comment]
Doing a file compare with the certificate text from the 'Received WU' compared with the text from the 'Sending dump report', there is extra text for "path" in the return data that looks like it probably shouldn't be there. It seems like it would change the "assignment" section signature:

(This is from the log file I attached the other day)
![image](https://user-images.githubusercontent.com/20870053/210135307-6c72d641-b3a3-4938-a49b-05f280670fa1.png)
![image](https://user-images.githubusercontent.com/20870053/210134753-d310a401-9960-48f6-aaf4-eed3b609a2fc.png)

**Comment by jcoffland** [comment]
Nice catch!  I was not looking in the right place.  This exactly explains the behavior of the bug.  The first WU to access the core will get it's assignment data corrupted.  The `Core` class saves `path` in the same data structure.  Any subsequent usages of the core in the same run of the client will not be affected.  This will be fixed in the next release. Thanks!

**Comment by jcoffland** [comment]
Should be fixed by v8.1.7.  Reopen if not.
